### PR TITLE
Numpydoc-ify various functions

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

Sorry, this is a bit all-over-the-place. I started this a _long_ time ago, and in the interest of it not becoming further out of date, I decided to clean it up a bit and post it. Unfortunately, it's a bit of a hodge-podge of different things right now.

## PR Checklist

- [N/A] Has Pytest style unit tests
- [x] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [N/A] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [N/A] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [N/A] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/dates.py [hunk 1]
```diff
@@ -607,8 +607,13 @@ class IndexDateFormatter(ticker.Formatter):
...
         """
-        *t* is a sequence of dates (floating point days).  *fmt* is a
-        `~datetime.datetime.strftime` format string.
+        Parameters
+        ----------
+        t:
+            A sequence of dates (floating point days).
+        fmt:
+            A `~datetime.datetime.strftime` format string.
+
```
Comment (anntzer) [inline lib/matplotlib/dates.py:616]: prefer no blank line here

File: lib/matplotlib/legend.py [hunk 2]
```diff
@@ -978,13 +986,16 @@ def _get_anchored_bbox(self, loc, bbox, parentbbox, renderer):
...
-        - loc: a location code in range(1, 11).
-          This corresponds to the possible values for self._loc, excluding
-          "best".
+        Parameters
+        ----------
+        loc : int
+            A location code in range(1, 11). This corresponds to the possible
+            values for self._loc, excluding "best".
+        bbox : `~matplotlib.transforms.Bbox`
+            bbox to be placed, display coodinate units.
```
Comment (anntzer) [inline lib/matplotlib/legend.py:995]: coodinate: typo
also be consistent with immediately below (e.g. ", in display coordinates.") for both.

File: lib/matplotlib/lines.py [hunk 3]
```diff
@@ -1493,8 +1493,13 @@ def process_selected(self, ind, xs, ys):
         Default "do nothing" implementation of the
         :meth:`process_selected` method.
 
-        *ind* are the indices of the selected vertices.  *xs* and *ys*
-        are the coordinates of the selected vertices.
+        Parameters
+        ----------
+        ind:
```
Comment (anntzer) [inline lib/matplotlib/lines.py:1498]: no colons

File: lib/matplotlib/offsetbox.py [hunk 4]
```diff
@@ -142,10 +142,17 @@ def _get_aligned_offsets(hd_list, height, align="baseline"):
     *mode*. xdescent is analogous to the usual descent, but along the
     x-direction. xdescent values are currently ignored.
 
-    *hd_list* : list of (width, xdescent) of boxes to be aligned.
-    *sep* : spacing between boxes
-    *height* : Intended total length. None if not used.
-    *align* : align mode. 'baseline', 'top', 'bottom', or 'center'.
+    Parameters
+    ----------
+    hd_list:
```
Comment (anntzer) [inline lib/matplotlib/offsetbox.py:147]: same

File: lib/matplotlib/sankey.py [hunk 5]
```diff
@@ -168,15 +170,15 @@ def _arc(self, quadrant=0, cw=True, radius=1, center=(0, 0)):
         Return the codes and vertices for a rotated, scaled, and translated
         90 degree arc.
 
-        Optional keyword arguments:
+        Other Parameters
+        ----------------
+        quadrant : int, {0, 1, 2, 3}
+            Uses 0-based indexing (0, 1, 2, or 3)
```
Comment (anntzer) [inline lib/matplotlib/sankey.py:176]: end with full stops

File: lib/matplotlib/dates.py [hunk 6]
```diff
@@ -607,8 +607,13 @@ class IndexDateFormatter(ticker.Formatter):
 
     def __init__(self, t, fmt, tz=None):
         """
-        *t* is a sequence of dates (floating point days).  *fmt* is a
-        `~datetime.datetime.strftime` format string.
+        Parameters
+        ----------
+        t:
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply lib/matplotlib/dates.py:612]: Added types instead.

File: lib/matplotlib/sankey.py [hunk 7]
```diff
@@ -168,15 +169,14 @@ def _arc(self, quadrant=0, cw=True, radius=1, center=(0, 0)):
...
-          ===============   ==========================================
-          Keyword           Description
-          ===============   ==========================================
-          *quadrant*        uses 0-based indexing (0, 1, 2, or 3)
-          *cw*              if True, clockwise
-          *center*          (x, y) tuple of the arc's center
-          ===============   ==========================================
+        Other Parameters
+        ----------------
+        quadrant : int, {0, 1, 2, 3}
```
Comment (anntzer) [inline lib/matplotlib/sankey.py:174]: I guess the only acceptable values are 0, 1, 2, 3, not any int.
(Also perhaps state the defaults.)

Comment (QuLogic) [reply lib/matplotlib/sankey.py:174]: I'm not really sure what 0, 1, 2, 3 mean, and I'm getting the feeling the optimization is not really worth it and we should just replace this function with `Path.arc`... But that would be a separate PR.

File: lib/matplotlib/lines.py [hunk 8]
```diff
@@ -1493,8 +1493,12 @@ def process_selected(self, ind, xs, ys):
         Default "do nothing" implementation of the
         :meth:`process_selected` method.
 
-        *ind* are the indices of the selected vertices.  *xs* and *ys*
-        are the coordinates of the selected vertices.
+        Parameters
+        ----------
+        ind
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply lib/matplotlib/lines.py:1498]: Checked and this one is a list.

**Comment by tacaswell** [comment]
Anyone can merge on green.

**Pull request merged**